Userland/Libraries/LibWeb/Layout/TableFormattingContext.cpp

```diff
@@ -559,6 +559,11 @@ CSSPixels TableFormattingContext::compute_capmin()
     return capmin;
 }
 
+static bool width_is_auto_relative_to_state(CSS::Size const& width, LayoutState::UsedValues const& state)
+{
+    return width.is_auto() || (width.contains_percentage() && !state.has_definite_width());
+}
```

Comment:
is that possible to use [FormattingContext::should_treat_width_as_auto](https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibWeb/Layout/FormattingContext.cpp#L1615) instead?

Reply:
Unfortunately no, that one takes available space and here we resolve relative to the parent container size instead.

Reply:
but you can can construct available space and pass it to the `should_treat_width_as_auto`, right?

Reply:
We only have a `LayoutState::UsedValues`, it doesn't look like we have an appropriate constructor for `AvailableSize` which would be useful for that. If we'd have to branch between `AvailableSize::make_indefinite()` and `AvailableSize::make_definite` anyway, would you still like it better?